### PR TITLE
InitiateSystemShutdown, ExitWindowsEx

### PR DESCRIPTION
**Pull request by Green7**

### CODE REVIEW

File: system.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+package gowin32
+
+import (
+	"syscall"
+
+	"github.com/winlabs/gowin32/wrappers"
+)
+
+func InitiateSystemShutdown(machineName string, message string, timeout int, forceAppsClosed bool, rebootAfterShutdown bool) error {
+	return wrappers.InitiateSystemShutdown(syscall.StringToUTF16Ptr(machineName), syscall.StringToUTF16Ptr(message), uint32(timeout), forceAppsClosed, rebootAfterShutdown)
```
Comment (dgolub) [inline system.go]: Should wrap with `NewWindowsError` if it fails.  Also, given the length of this, it probably wouldn't hurt to split it with each parameter on a separate line.

File: wrappers/reason.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	SHTDN_REASON_MINOR_NETWORK_CONNECTIVITY  = 0x00000014
+	SHTDN_REASON_MINOR_WMI                   = 0x00000015
+	SHTDN_REASON_MINOR_SERVICEPACK_UNINSTALL = 0x00000016
+	SHTDN_REASON_MINOR_HOTFIX_UNINSTALL      = 0x00000017
+	SHTDN_REASON_MINOR_SECURITYFIX_UNINSTALL = 0x00000018
+	SHTDN_REASON_MINOR_MMC                   = 0x00000019
+	SHTDN_REASON_MINOR_SYSTEMRESTORE         = 0x0000001a
+	SHTDN_REASON_MINOR_TERMSRV               = 0x00000020
+	SHTDN_REASON_MINOR_DC_PROMOTION          = 0x00000021
+	SHTDN_REASON_MINOR_DC_DEMOTION           = 0x00000022
```
Comment (dgolub) [inline wrappers/reason.go]: It appears that `SHTDN_REASON_MAJOR_NONE`, `SHTDN_REASON_MINOR_NONE`, `SHTDN_REASON_MINOR_SYSTEMRESTORE`, `SHTDN_REASON_MINOR_DC_PROMOTION`, and `SHTDN_REASON_MINOR_DC_DEMOTION` are undocumented, so let's remove them from here.  See https://docs.microsoft.com/en-us/windows/desktop/shutdown/system-shutdown-reason-codes.

File: wrappers/winuser.go [hunk 3]
```diff
@@ -106,6 +106,18 @@ const (
...
 )
 
+const (
+	EWX_LOGOFF          = 0
+	EWX_SHUTDOWN        = 0x00000001
+	EWX_REBOOT          = 0x00000002
+	EWX_FORCE           = 0x00000004
+	EWX_POWEROFF        = 0x00000008
+	EWX_FORCEIFHUNG     = 0x00000010
+	EWX_QUICKRESOLVE    = 0x00000020
```
Comment (dgolub) [inline wrappers/winuser.go]: `EWX_QUICKRESOLVE` also appears to be undocumented.  See https://docs.microsoft.com/en-us/windows/desktop/api/winuser/nf-winuser-exitwindowsex.

File: wrappers/winuser.go [hunk 4]
```diff
@@ -235,6 +248,23 @@ func CloseDesktop(desktop syscall.Handle) error {
 	return nil
 }
 
+func ExitWindowsEx(flags uint, reason uint32) error {
```
Comment (dgolub) [inline wrappers/winuser.go]: [Nitpick] Flags should also be `uint32`.

**Comment by Green7** [comment]
Hello. How about this PR ? Does something need improvement?

**Pull request merged**